src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -3,46 +3,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { IncomingMessage } from 'http'
-import * as request from 'request'
 import { getLogger, Logger } from '../logger'
+import { getResponseFromGetRequest } from '../utilities/requestUtils'
 import { ResourceFetcher } from './resourcefetcher'
 
 export class HttpResourceFetcher implements ResourceFetcher {
     private readonly logger: Logger = getLogger()
 
-    public constructor(private readonly url: string) {}
+    public constructor(private readonly url: string, private readonly friendlyName?: string) {}
 
     /**
      * Returns the contents of the resource, or undefined if the resource could not be retrieved.
      */
     public async get(): Promise<string | undefined> {
         try {
-            this.logger.verbose(`Loading resource from ${this.url}`)
+            this.logger.verbose(`Loading ${this.logText()}`)
 
             const contents = await this.loadFromUrl()
 
-            this.logger.verbose(`Finished loading resource from ${this.url}`)
+            this.logger.verbose(`Finished loading ${this.logText()}`)
 
             return contents
         } catch (err) {
-            this.logger.error(`Error loading resource from ${this.url}: %O`, err as Error)
+            this.logger.error(`Error loading ${this.logText()}: %O`, err as Error)
 
             return undefined
         }
     }
 
+    // TODO: Are there cases where we don't mind the URL?
+    // Safer to do it this way assuming others use this library.
+    private logText(): string {
+        return this.friendlyName ?? 'resource from external URL'
```

Comment:
what does "external" mean here? is there anything preventing use of an "internal" URL with this module?

Reply:
Just a filler word. If you have a better phrase feel free to suggest.

Reply:
i would remove "external"

Reply:
Already done